src/main/java/org/sagebionetworks/web/client/widget/table/v2/TotalVisibleResultsWidget.java

```diff
@@ -0,0 +1,91 @@
+package org.sagebionetworks.web.client.widget.table.v2;
+
+import static org.sagebionetworks.repo.model.table.QueryOptions.BUNDLE_MASK_QUERY_COUNT;
+import static org.sagebionetworks.web.client.widget.asynch.AsynchronousProgressWidget.WAIT_MS;
+import static org.sagebionetworks.web.client.widget.table.v2.results.QueryBundleUtils.getDefaultQuery;
+
+import org.sagebionetworks.repo.model.asynch.AsynchronousJobStatus;
+import org.sagebionetworks.repo.model.asynch.AsynchronousResponseBody;
+import org.sagebionetworks.repo.model.table.Dataset;
+import org.sagebionetworks.repo.model.table.QueryBundleRequest;
+import org.sagebionetworks.repo.model.table.QueryResultBundle;
+import org.sagebionetworks.repo.model.table.View;
+import org.sagebionetworks.web.client.PopupUtilsView;
+import org.sagebionetworks.web.client.widget.asynch.AsynchronousJobTracker;
+import org.sagebionetworks.web.client.widget.asynch.UpdatingAsynchProgressHandler;
+import org.sagebionetworks.web.shared.asynch.AsynchType;
+
+import com.google.gwt.user.client.ui.IsWidget;
+import com.google.gwt.user.client.ui.Widget;
+import com.google.inject.Inject;
+
+public class TotalVisibleResultsWidget implements IsWidget {
+
+	private TotalVisibleResultsWidgetView view;
+	private Integer totalNumberOfResults;
+	private Integer totalNumberOfVisibleResults;
+	private PopupUtilsView popupUtils;
+	private AsynchronousJobTracker asyncJobTracker;
+
+	private static final String DATASETS_HELP = "Files may be unavailable because you do not have permission to see them, they have been deleted, or the Dataset has been misconfigured.";
+
+	@Inject
+	public TotalVisibleResultsWidget(TotalVisibleResultsWidgetView view, PopupUtilsView popupUtils, AsynchronousJobTracker asyncJobTracker) {
+		this.view = view;
+		this.popupUtils = popupUtils;
+		this.asyncJobTracker = asyncJobTracker;
+	}
+
+	@Override
+	public Widget asWidget() {
+		return view.asWidget();
+	}
+
+	public void configure(View viewEntity) {
+		// Currently, the only case where we can reliably get this info is for the current version of a Dataset.
+		// Other cases will need a new service, tracked by PLFM-7046
+		if (viewEntity instanceof Dataset && viewEntity.getIsLatestVersion()) {
+			this.totalNumberOfResults = ((Dataset) viewEntity).getItems().size();
+			view.setTotalNumberOfResults(totalNumberOfResults);
+			view.setHelpMarkdown(DATASETS_HELP);
```

Comment:
Is the idea that this help text will change in the future, depending on the Table type?  It looks like this is unnecessary today (since it's currently initialized to this help markdown text)

Reply:
Correct